source/common/config/delta_subscription_state.cc

```diff
@@ -124,6 +134,29 @@ void DeltaSubscriptionState::handleEstablishmentFailure() {
 
 envoy::api::v2::DeltaDiscoveryRequest DeltaSubscriptionState::getNextRequestAckless() {
   envoy::api::v2::DeltaDiscoveryRequest request;
+  if (!any_request_sent_yet_in_current_stream_) {
+    populateDiscoveryRequest(request);
+  } else if (!aliases_added_.empty()) {
+    populateDiscoveryRequestWithAliases(request);
+  } else {
+    populateDiscoveryRequest(request);
+  }
+
+  request.set_type_url(type_url_);
+  request.mutable_node()->MergeFrom(local_info_.node());
+  return request;
+}
+
+//
+void DeltaSubscriptionState::populateDiscoveryRequestWithAliases(
```

Comment:
Is this supposed to then call through to populateDiscoveryRequest after doing its stuff? As it is, it looks like when there are aliases, nothing but the aliases get sent.